### ISSUE TITLE
data_backup

### ISSUE THREAD

**Issue by ziandzivan** [Magisk-Modules-Alt-Repo/submission#162]
**ID**: data_backup
**Name**: data backup
**Description**: Script of creating User Data backup from booted system with root (subsequent restoring by data_restore.zip script) https://github.com/ziandzivan/data_backup#readme
**Link**: https://github.com/ziandzivan/data_backup
**Source code**: https://github.com/ziandzivan/data_backup/archive/refs/tags/v2.11.zip

**Comment by Atrate** [comment]
Where can I find the source code for the binary embedded in the script?

**Comment by ziandzivan** [comment]
There are all binaries in /system)/bin/data_backup.sh. They attached in the end of the script.
According script code means address positions :
`dd if="$scr_arg0" bs=1024 skip=5 count=753 > "${D_bin}/unpack"` - 7za utility (not zipped). 
`dd if="$scr_arg0" bs=1024 skip=$((753 + 5)) count=2 > "${D_temp}/utl"` - few backup Utility Functions (zipped)
`dd if="$scr_arg0" bs=1024 skip=$((753 + 5 + 2)) > "${D_temp}/bins"` - zipped binaries: busybox, bash, GNU tar, service shell script: new_vlm.
Let me know if you would like me to extract and attach something here ?
Thanks

**Comment by Atrate** [comment]
Why do you embed those binaries in the script instead of just including them in `/system/bin` as-is?

**Comment by ziandzivan** [comment]
Why so special binaries, utilities, scripts should always hang with data_backup.sh in the device ?
They must extracted and used only at running data backup script, after backuping must be removed.

**Comment by Atrate** [comment]
It's a Magisk module, so they will be removed when the user decides to remove the module anyways.

**Comment by ziandzivan** [comment]
No. The script must be always installed by module and ever workable even the display is crashed to get last data backup.
the attachments must be installed when the script is started and removed after exit.
First versions of my dat_backups since of 2018 year. No any necessity changing the algorithm for users.

**Comment by Atrate** [comment]
> The script must be always installed by module

So do you copy the script to a separate location or do you just want the users to always have the module installed?

>  Let me know if you would like me to extract and attach something here ?

I'd like it if you kept some place (separate repo, wiki) with all the extracted files, so they can always be viewed without the hassle of getting them out of the script.

**Comment by ziandzivan** [comment]
> So do you copy the script to a separate location or do you just want the users to always have the module installed?
How it should be correctly already done - the module and the script show this.

> so they can always be viewed without the hassle
It open for anyone. It is a method of attached any addition without keep them active in memory to avoid any troubles with other user utilities, busybox and so on. This used many developers, for example look at some first versions of Magisk (at about 20.4).

No problem. I will open new repro for these additions.

**Comment by ziandzivan** [comment]
here are the attaches: https://github.com/ziandzivan/attachments_of_data_backup
thanks

**Comment by Atrate** [comment]
> here are the attaches: https://github.com/ziandzivan/attachments_of_data_backup thanks

Thanks. Where can I find the source code of `backup`? 

Additionally, please add:
- A link to https://github.com/ziandzivan/attachments_of_data_backup to your main repo's README
- Links to the source code of all compiled binaries used, as per MMAR rules

**Comment by ziandzivan** [comment]
> Where can I find the source code of `backup`?
It is GNU tar v1.34, download the source code :  https://www.gnu.org/software/tar/
Manual: https://www.gnu.org/software/tar/manual/html_node/index.html#Top

**Comment by ziandzivan** [comment]
I have to add to README source codes of ALL utilities that my shell script uses ? such as tar, echo, mkdir, rm, dd, ... and so on ?
I really do not understand...
Please explain (?)

**Comment by Atrate** [comment]
> You must provide the source code for any executables or APK files included in your submission

You have a total of **4** compiled binaries used in your module (busybox, bash, tar (backup) and 7za, all that's needed is to provide the links to their source code.
